From cc28dc80d5809d1573ba1d615b4356c2292ee95c Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Thu, 2 Jan 2025 15:26:29 +0800
Subject: [PATCH 155/210] mtk: hostapd: fix hostapd crash when building per-sta
 profile

Do not build the profile for links who are not ready.
If a link is not ready, hostapd will crash because hapd->iface->current_mode
is null in hostapd_eid_ht_operation.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 src/ap/beacon.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/ap/beacon.c b/src/ap/beacon.c
index ca6405e24..2bb99aeae 100644
--- a/src/ap/beacon.c
+++ b/src/ap/beacon.c
@@ -3426,6 +3426,8 @@ int ieee802_11_set_beacon(struct hostapd_data *hapd)
 
 	/* Generate per STA profiles for each affiliated APs */
 	for_each_mld_link(link_bss, hapd) {
+		if (!link_bss->started)
+			continue;
 		hostapd_gen_per_sta_profiles(link_bss);
 
 		if (link_bss->eht_mld_bss_critical_update == BSS_CRIT_UPDATE_FLAG &&
-- 
2.45.2

